src/pages/Bridge/index.tsx

```diff
@@ -81,83 +159,268 @@ const BridgePage: React.FC = () => {
+
   const handleChainFieldChange = useCallback(
-    ({ fromChainId, toChainId }: { fromChainId?: ChainId; toChainId?: ChainId }) => {
+    ({ newFromChainId, newToChainId }: { newFromChainId?: ChainId; newToChainId?: ChainId }) => {
       const formValues = getValues();
 
-      if (fromChainId && toChainId) {
+      if (newFromChainId && newToChainId) {
         switchChain({
-          chainId: fromChainId,
+          chainId: newFromChainId,
           callback: () => {
-            setValue('fromChainId', fromChainId);
-            setValue('toChainId', toChainId);
+            setValue('fromChainId', newFromChainId);
+            setValue('toChainId', newToChainId);
           },
         });
         return;
       }
 
-      if (fromChainId) {
+      if (newFromChainId) {
         switchChain({
-          chainId: fromChainId,
+          chainId: newFromChainId,
           callback: () => {
-            setValue('fromChainId', fromChainId);
+            setValue('fromChainId', newFromChainId);
 
-            if (formValues.toChainId === fromChainId) {
+            if (formValues.toChainId === newFromChainId) {
               setValue('toChainId', formValues.fromChainId);
             }
           },
         });
         return;
       }
 
-      if (toChainId && formValues.fromChainId === toChainId) {
+      if (newToChainId && formValues.fromChainId === newToChainId) {
         switchChain({
           chainId: formValues.toChainId,
           callback: () => {
             setValue('fromChainId', formValues.toChainId);
-            setValue('toChainId', toChainId);
+            setValue('toChainId', newToChainId);
           },
         });
         return;
       }
 
-      if (toChainId) {
-        setValue('toChainId', toChainId);
+      if (newToChainId) {
+        setValue('toChainId', newToChainId);
       }
     },
     [setValue, switchChain, getValues],
   );
 
+  // check if there is enough native token to pay the estimated fee
+  useEffect(() => {
+    const val =
+      bridgeEstimatedFeeMantissa.gt(0) &&
+      nativeTokenBalanceMantissa.lte(bridgeEstimatedFeeMantissa);
+
+    setValue('doesNotHaveEnoughNativeFunds', val, { shouldValidate: true });
+  }, [bridgeEstimatedFeeMantissa, nativeTokenBalanceMantissa, setValue]);
+
+  // check if amount of XVS informed by the user is over the max single transaction limit in USD
+  useEffect(() => {
+    const val =
+      !!getXvsBridgeStatusData &&
+      xvsAmountUsd.gt(getXvsBridgeStatusData.maxSingleTransactionLimitUsd);
+
+    setValue('isSingleTransactionLimitExceeded', val, { shouldValidate: true });
+  }, [getXvsBridgeStatusData, setValue, xvsAmountUsd]);
+
+  // check if the global daily limit of XVS transferred was reached
+  useEffect(() => {
+    const val = xvsAmountUsd.gt(remainingXvsDailyLimitUsd);
+
+    setValue('isDailyTransactionLimitExceeded', val, { shouldValidate: true });
+  }, [remainingXvsDailyLimitUsd, setValue, xvsAmountUsd]);
```

Comment:
Following my previous comment: this logic should be written into the validation schema of the form.